# FINAL PROJECT ON DATA SCIENCE UC-SAN DIEGO COURSE

In [46]:
import numpy as np
import pandas as  pd
import matplotlib.pyplot as plt
import sqlite3
from sklearn.tree import DecisionTreeRegressor
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error
from math import sqrt

In [8]:
cnx = sqlite3.connect('database.sqlite')
player = pd.read_sql_query('SELECT * from Player',cnx)
data_player = pd.read_sql_query('SELECt * from Player_Attributes',cnx)

In [12]:
len(player)

11060

In [15]:
data_player = data_player[:11060]
len(data_player)

11060

In [24]:
data = player.merge(data_player,on='id')

In [54]:
data.head(2)


id         player_name  player_fifa_api_id_x             birthday  height  \
0   1  Aaron Appindangoye                218353  1992-02-29 00:00:00  182.88   
1   2     Aaron Cresswell                189615  1989-12-15 00:00:00  170.18   

   weight  player_fifa_api_id_y  player_api_id_y                 date  \
0     187                218353           505942  2016-02-18 00:00:00   
1     146                218353           505942  2015-11-19 00:00:00   

   overall_rating     ...       vision penalties marking standing_tackle  \
0            67.0     ...         54.0      48.0    65.0            69.0   
1            67.0     ...         54.0      48.0    65.0            69.0   

   sliding_tackle  gk_diving  gk_handling  gk_kicking  gk_positioning  \
0            69.0        6.0         11.0        10.0             8.0   
1            69.0        6.0         11.0        10.0             8.0   

   gk_reflexes  
0          8.0  
1          8.0  

[2 rows x 47 columns]

In [27]:
data.columns

Index(['id', 'player_name', 'player_fifa_api_id_x', 'birthday', 'height',
       'weight', 'player_fifa_api_id_y', 'player_api_id_y', 'date',
       'overall_rating', 'potential', 'preferred_foot', 'attacking_work_rate',
       'defensive_work_rate', 'crossing', 'finishing', 'heading_accuracy',
       'short_passing', 'volleys', 'dribbling', 'curve', 'free_kick_accuracy',
       'long_passing', 'ball_control', 'acceleration', 'sprint_speed',
       'agility', 'reactions', 'balance', 'shot_power', 'jumping', 'stamina',
       'strength', 'long_shots', 'aggression', 'interceptions', 'positioning',
       'vision', 'penalties', 'marking', 'standing_tackle', 'sliding_tackle',
       'gk_diving', 'gk_handling', 'gk_kicking', 'gk_positioning',
       'gk_reflexes'],
      dtype='object')

In [68]:
data.isnull().values.any(axis=1)
data = data.dropna()
#thus our data frame is clean

#### Aim of our analysis is to predict the Overall Rating of a Player based on his Skills


In [69]:
skills = ['height','weight','potential', 'crossing', 'finishing', 'heading_accuracy',
       'short_passing', 'volleys', 'dribbling', 'curve', 'free_kick_accuracy',
       'long_passing', 'ball_control', 'acceleration', 'sprint_speed',
       'agility', 'reactions', 'balance', 'shot_power', 'jumping', 'stamina',
       'strength', 'long_shots', 'aggression', 'interceptions', 'positioning',
       'vision', 'penalties', 'marking', 'standing_tackle', 'sliding_tackle',
       'gk_diving', 'gk_handling', 'gk_kicking', 'gk_positioning',
       'gk_reflexes']

In [70]:
y = data['overall_rating']
type(y)

pandas.core.series.Series

In [71]:
x= data[skills]
type(x)

pandas.core.frame.DataFrame

In [72]:
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.4,random_state=0)

In [73]:
type(x_train)

pandas.core.frame.DataFrame

In [74]:
regressor = LinearRegression()
regressor.fit(x_train,y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=1, normalize=False)

In [75]:
y_prediction = regressor.predict(x_test)

In [78]:
y_prediction[:5]

array([ 60.54154109,  66.6034816 ,  75.02100016,  76.56463146,  65.65052243])

In [79]:
y_test[:5]

6934    61.0
2168    64.0
159     81.0
9569    77.0
1520    62.0
Name: overall_rating, dtype: float64

In [80]:
RMSE = sqrt(mean_squared_error(y_true=y_test,y_pred=y_prediction))
print(RMSE)

2.807321690667002


#### thus we predicted the Overall Ratings of a player with an Error of 2.8%